hail/src/main/scala/is/hail/expr/ir/TableWriter.scala

```diff
@@ -433,7 +436,115 @@ case class TableTextWriter(
   exportType: String = ExportType.CONCATENATED,
   delimiter: String
 ) extends TableWriter {
+
   def apply(ctx: ExecuteContext, tv: TableValue): Unit = tv.export(ctx, path, typesFile, header, exportType, delimiter)
+
+  override def canLowerEfficiently: Boolean = true
+  override def lower(ctx: ExecuteContext, ts: TableStage, t: TableIR, r: RTable, relationalLetsAbove: Map[String, IR]): IR = {
+    if (exportType == ExportType.PARALLEL_COMPOSABLE)
+      throw new LowererUnsupportedOperation(s"cannot lower parallel_composable export")
```

Comment:
Probably you want `canLowerEfficiently` to only be true if `exportType != ExportType.PARALLEL_COMPOSABLE`. Otherwise the thing that looks at pipelines and decides if it can lower them instead of doing old Spark execution model is going to decide that it can, only to error when it tries.

Reply:
done